### PR TITLE
Incorrect row comparison for tpch queries in benchmarks

### LINKED ISSUE #5783: Incorrect row comparison for tpch queries in benchmarks
### Describe the bug

We run tpch queries in benchmarks and compare query output with expected results. But currently `verify_query` compares rows between actual and expected incorrectly.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #5783.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -1106,7 +1106,7 @@ mod ci {
             let actual_row = &actual_vec[i];
             assert_eq!(expected_row.len(), actual_row.len());
 
-            for j in 0..expected.len() {
```
Comment (viirya) [inline benchmarks/src/bin/tpch.rs:1109]: e.g., If `expected` contains only one record batch, we only compare first column between rows.

Comment (viirya) [reply benchmarks/src/bin/tpch.rs:1109]: Not sure if any existing tests will fail by this.

File: benchmarks/src/bin/tpch.rs [hunk 2]
```diff
@@ -1118,6 +1118,20 @@ mod ci {
...
+                        let tolerance = 0.1;
+                        if ((l - r) as f64 / 10_i32.pow(*s as u32) as f64).abs()
+                            > tolerance
+                        {
+                            panic!(
+                                "Expected: {}; Actual: {}; Tolerance: {}",
+                                l, r, tolerance
+                            )
+                        }
+                    }
```
Comment (viirya) [inline benchmarks/src/bin/tpch.rs:1134]: This is necessary to pass `ci::verify_q9` after this change. There is a slight (`0.01`) difference.

File: benchmarks/src/bin/tpch.rs [hunk 3]
```diff
@@ -1118,6 +1118,20 @@ mod ci {
                             )
                         }
                     }
+                    (
+                        ScalarValue::Decimal128(Some(l), _, s),
+                        ScalarValue::Decimal128(Some(r), _, _),
+                    ) => {
+                        let tolerance = 0.1;
```
Comment (alamb) [inline benchmarks/src/bin/tpch.rs:1125]: tolerance is already defined above in the `float` path -- maybe we could make it a constant or something. Very minor

**Review by alamb — APPROVED** [review]

**Pull request merged**